graylog2-web-interface/src/components/pipelines/PipelineListItem.tsx

```diff
@@ -0,0 +1,160 @@
+ */
+import React from 'react';
+import styled, { css } from 'styled-components';
+
+import { isPermitted } from 'util/PermissionsMixin';
+import { Link, LinkContainer } from 'components/common/router';
+import Routes from 'routing/Routes';
+import { CounterRate, MetricContainer } from 'components/metrics';
+import PipelineConnectionsList from 'components/pipelines/PipelineConnectionsList';
+import { Button, Label } from 'components/bootstrap';
+import type { PipelineType } from 'stores/pipelines/PipelinesStore';
+import type { PipelineConnectionsType } from 'stores/pipelines/PipelineConnectionsStore';
+import type { Stream } from 'logic/streams/types';
+import { defaultCompare as naturalSort } from 'logic/DefaultCompare';
+import useCurrentUser from 'hooks/useCurrentUser';
+import useGetPermissionsByScope from 'hooks/useScopePermissions';
+
+import ButtonToolbar from '../bootstrap/ButtonToolbar';
+import { Spinner } from '../common';
+
+type Props = {
+  pipeline: PipelineType;
+  pipelines: Array<PipelineType>;
+  connections: Array<PipelineConnectionsType>;
+  streams: Array<Stream>;
+  onDeletePipeline: () => void;
+};
+const PipelineNameTD = styled.td`
+  max-width: 300px;
+  overflow-x: hidden;
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  width: 300px;
+`;
+const StreamListTD = styled.td`
+  max-width: 150px;
+  width: 150px;
+  word-wrap: break-word;
+`;
+
+const PipelineStage = styled.div<{ $idle?: boolean }>(
+  ({ $idle, theme }) => css`
+    border: 1px solid ${theme.colors.gray[$idle ? '50px' : '70px']};
+    border-radius: 4px;
+    display: inline-block;
+    margin-right: 15px;
+    padding: 20px;
+    text-align: center;
+    width: 120px;
+    background-color: ${$idle
+      ? theme.utils.colorLevel(theme.colors.global.contentBackground, 10)
+      : theme.colors.global.contentBackground};
+  `,
+);
+const DefaultLabel = styled(Label)`
+  display: inline-flex;
+  margin-left: 5px;
+  vertical-align: inherit;
+`;
+
+const PipelineListItem = ({ pipeline, pipelines, connections, streams, onDeletePipeline }: Props) => {
+  const currentUser = useCurrentUser();
+  const { loadingScopePermissions, scopePermissions } = useGetPermissionsByScope(pipeline);
+  const { id, title, description, stages } = pipeline;
+  const isManaged = scopePermissions && !scopePermissions?.is_mutable;
+  const isNotDeletable = scopePermissions && !scopePermissions?.is_deletable;
+  const _formatConnectedStreams = (streamsUsingPipeline: Array<Stream>) =>
+    streamsUsingPipeline.map((s) => s.title).join(', ');
+  const _formatStages = () => {
+    const stageNumbers = stages.map((stage) => stage.stage);
+
+    return pipelines
+      .map(({ stages: pipelineStages }) => pipelineStages.map(({ stage }) => stage))
+      .reduce(
+        (usedStagesAcc: number[], pipelineStages: number[]) =>
+          // Concat stages in a single array removing duplicates
+          Array.from(new Set([...usedStagesAcc, ...pipelineStages])),
```

Comment:
Maybe we can extract this to a function `getStagesWithoutDuplicates` then it's self explanatory and we don't need the comment?